### PR TITLE
WIP implement pagination in the app

### PR DESCRIPTION
**Pull request by nikit19**

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/BoundaryCallback.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import android.util.Log
+
+
+class BoundaryCallback(
+        private val eventDao: EventDao,
+        private val service: EventApi
+) : PagedList.BoundaryCallback<Event>() {
+
+    companion object {
+        private const val NETWORK_PAGE_SIZE = 50
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/BoundaryCallback.kt]: Use 20

File: app/src/main/java/org/fossasia/openevent/general/event/BoundaryCallback.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    // keep the last requested page. When the request is successful, increment the page number.
+    private var lastRequestedPage = 1
+
+    private val _networkErrors = MutableLiveData<String>()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/BoundaryCallback.kt]: No underscore in names

**Comment by iamareebjamal** [comment]
Links are present in the request when done. One of the link is of next page. Take a look

**Comment by nikit19** [comment]
@iamareebjamal thanks just saw that.
I am not able to come up with the logic required for BoundaryCallback
I think pagination should work when BoundaryCallback is written correctly.
(if everything else is correct)

**Comment by nikit19** [comment]
Closing this PR as it is not required at the moment. Will send a new PR if it is needed

**Comment by nikit19** [comment]
Actually I thought since now we are using `filter` everywhere we were getting all the events. But that is not the case. I will send a PR for this then as soon as I complete it

**Pull request closed without merging**